matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -115,21 +117,16 @@ public MapMatching(GraphHopper graphHopper, PMap hints) {
         final double headingTimePenalty = hints.getDouble(Parameters.Routing.HEADING_PENALTY, Parameters.Routing.DEFAULT_HEADING_PENALTY);
         uTurnDistancePenalty = headingTimePenalty * PENALTY_CONVERSION_VELOCITY;
 
-        boolean disableCH = hints.getBool(Parameters.CH.DISABLE, false);
-        if (graphHopper.getCHPreparationHandler().isEnabled() && disableCH && !graphHopper.getRouterConfig().isCHDisablingAllowed())
-            throw new IllegalArgumentException("Disabling CH is not allowed");
-
-        if (graphHopper.getCHPreparationHandler().isEnabled() && !disableCH) {
-            ch = true;
-            routingGraph = graphHopper.getGraphHopperStorage().getCHGraph(profile.getName());
-            if (routingGraph == null)
-                throw new IllegalArgumentException("Cannot find CH preparation for the requested profile: '" + profile + "'" +
-                        "\nYou can try disabling CH using " + Parameters.CH.DISABLE + "=true" +
-                        "\navailable CH profiles: " + graphHopper.getGraphHopperStorage().getCHGraphNames());
+        boolean disableLM = hints.getBool(Parameters.Landmark.DISABLE, false);
```

Comment:
So far we could enable Dijkstra using `ch.disable=true`, but now we would need `lm.disable=true` and following this logic later it might be `ch.disable=true` again... Should we better disable LM/CH no matter which of the flags is given? Or should there be some kind of extra flag like `use_large_distance_routing` or similar? Or should we remove the flag altogether here (and better implement the heuristic that chooses to use Dijkstra when the point distance is below some threshold)?

Reply:
We could even remove this feature completely. Or what would be the use case? Like a different weighting (like short_fastest) or avoid=ferry per request? When I think about it then this would make a matching difference only for queries locations that are far away from each other where the map matching has to guess lots of things in-between and then using Dijkstra seems to be rather ugly :) (?)

Reply:
> using Dijkstra seems to be rather ugly

Yes for sparse data (long distances between gpx points) map matching can get very slow when using Dijkstra (its bad in a worst case scenario), so LM/CH is the better default. Switching to Dijkstra would be beneficial in case one knows that the gpx points are dense. Like I said, the idea solution would be choosing the appropriate algorithm internally. Keeping the flag could still be useful for debugging/testing.

Reply:
> Switching to Dijkstra would be beneficial in case one knows that the gpx points are dense

Yes, and in those cases it is unlikely that the flexibility of Dijkstra is really required because the points already force the route on a certain road (?)

For debugging we can surely keep but it should be no problem to remove the old CH disable parameter.

Reply:
Ah I meant in case the points are dense Dijkstra is (much) faster than CH, not just more flexible. I did a few measurements. I measured the map matching query time for different average gpx point distances.

**sparse**: average gpx point distance: 5km

|algorithm|query time (ms)|
|-|-|
|dijkstra|410|
|CH|76|
|LM|210|

**dense**: average gpx point distance: 100m

|algorithm|query time (ms)|
|-|-|
|dijkstra|28|
|CH|95|
|LM|36|

For the sparse setting CH is six times faster than Dijkstra and even more so if the distance was even larger. LM is of course still faster than Dijkstra but not as fast as CH obviously. In the dense setting CH is three times *slower* than Dijkstra and LM is also a bit slower than Dijkstra (but not so much).

Since LM still provides a good speed up compared to Dijkstra for sparse data while its still quite fast for the dense setting I think its a good default and actually there might be no real need to disable LM.

I think the most pragmatic solution would be accepting both the `ch.disable` and the `lm.disable` flags and use both to switch off whatever speed up technique we are or will be using. Its of course a bit inconsistent with the `/route` endpoint which is more strict with this regard, but since this is not documented/internal it does not really matter?

Reply:
Thanks for this!

> Its of course a bit inconsistent with the /route endpoint ... but since this is not documented/internal it does not really matter?

Ok, sounds good :+1:

Reply:
Ok so now we use LM by default (in case landmarks are available) and Dijkstra can be forced by setting either `ch.disable=true` (as before) or `lm.disable=true`.